pubsub/google/cloud/pubsub_v1/_gapic.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2017, Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+import functools
+
+
+def add_methods(SourceClass, blacklist=()):
```

Comment:
@jonparrott and I considered that but the problem is that the docs that you get as a result are distinctly inferior. I want `.. automodule` to actually know that the given class has these particular methods with these particular signatures.